### DISCUSSION TITLE
Ambiguous -C/--cache switch in man page?

### DISCUSSION THREAD

**Discussion by hug-fr** [osm2pgsql-dev/osm2pgsql#1588] [General]
In the [man page of osm2pgsql](https://osm2pgsql.org/doc/man/latest.html)  it is written that -C option should be used only when in slim mode 

> -C, --cache=NUM
Only for slim mode: Use up to NUM MB of RAM for caching nodes.

But it can be used alone and it helps to go over the 800MB threshold and be very useful to import big amount of data without using the slim mode which is very slow. I don't understand why it  is mentionned "only for slim mode". Shouldn't it be removed ?

**Comment by joto** [comment]
You can set the `-C, --cache` option in non-slim mode, it just isn't doing anything. Osm2pgsql ignores it in that case, because in non-slim mode all the nodes are in RAM anyway, so using the cache would use *more* memory without any actual benefit. In earlier versions of osm2pgsql this was actually the case: The cache was used, essentially storing some data twice.

I agree that the documentation is confusing and the manual on osm2pgsql.org is partially wrong, so that's definitely something we need to improve.

**Comment by hug-fr** [comment]
Thanks for your reply. As far as I understand, my "issue" is more likely to be a outdated version issue! Actually it was the one available on the debian stable repository I used and I didn't realized it was so old that the man page for this version is even no more avaible from osm2pgsql website. Sorry for the noise.

osm2pgsql -C 200000 -d db -U XXXXXX-W -H XXX.XXX.XXX.XXX -P 5432 europe-latest.osm.pbf
osm2pgsql version 0.96.0 (64 bit id space)

Password:
Using built-in tag processing pipeline
Using projection SRS 3857 (Spherical Mercator)
Setting up table: planet_osm_point
Setting up table: planet_osm_line
Setting up table: planet_osm_polygon
Setting up table: planet_osm_roads
Allocating memory for dense node cache
Allocating dense node cache in one big chunk
Allocating memory for sparse node cache
Sharing dense sparse
Node-cache: cache=200000MB, maxblocks=3200000*65536, allocation method=3

**Comment by lonvia** [comment]
Note that there are [backports](https://backports.debian.org/Instructions/) for osm2pgsql for Debian buster, which get you version 1.4.1.
